src/Lime/Protocol/Client/Channel.ts

```diff
@@ -21,16 +22,24 @@ export interface SessionChannel extends SessionListener {
   sendSession(session: Session): void;
 }
 
-abstract class Channel implements MessageChannel, CommandChannel, NotificationChannel, SessionChannel {
+export interface CommandProcessor extends CommandListener {
+  processCommand(command: Command): Promise<Command>;
+}
+
+abstract class Channel implements MessageChannel, CommandChannel, NotificationChannel, SessionChannel, CommandProcessor {
 
   private autoReplyPings: boolean;
   private autoNotifyReceipt: boolean;
 
+  commandTimeout = 6000;
+
   transport: Transport;
   remoteNode: string;
   localNode: string;
   sessionId: string;
   state: SessionState;
+  _commandResolves: {};
+  processCommandResolver: (command: Command) => any;

```

Comment:
This cannot be a single instance because a channel can have mulitple pending commands to process. I think we should rever the  implementation and make `commandResolves` receives the `resolve` functions instead of promises.